addons/stock/models/stock_picking.py

```diff
@@ -864,6 +864,10 @@ def write(self, vals):
         if vals.get('signature'):
             for picking in self:
                 picking._attach_sign()
+        if vals.get('picking_type_id'):
+            picking_type = self.env['stock.picking.type'].browse(vals.get('picking_type_id'))
+            for picking in self:
```

Comment:
You should only generate a new name if the value really changed. You are allowed to do:
```python
picking1.write({'picking_type_id': 5})
picking1.write({'picking_type_id': 5})
picking1.write({'picking_type_id': 5})
picking1.write({'picking_type_id': 5})
```
We don't want to increase the sequence and set a new name each time

Reply:
fixed and tested 👌